Add and Manage Creator "Support Levels"

briangershon opened the pull request:
## Tasks

- [x] Split Receipt and Support Level tests into separate files
- [x] Create Support Level entity
- [x] CRUD Operations for Creator's managing their Support Levels

## Acceptance Criteria

- [x] I can view my active Support Levels
- [x] I can delete an active Support Level
- [x] I can create and activate a new Support Level by specifying a title, a monthly price, a description and a benefit
- [x] Management of Creators support levels is decentralized, either on chain or through some other decentralized mechanism

Closes flobrij/issues/issues/6

briangershon committed:
Add SupportLevel account and initial test.

Split tests into two files:
  - Receipt
  - SupportLevel

briangershon committed:
Update Support Level. Add more tests and
placeholder tests.

akarys92 commented on programs/flobrij/src/lib.rs:
```diff
@@ -46,6 +46,67 @@ pub mod flobrij {
+    pub fn set_supportlevel(
+        ctx: Context<SetSupportLevel>,
+        title: String,
+        monthly_price: u64,
```
I think we should make this just "price" and then add another field to set how long a given  support level is active for. We can make it monthly in the front end, but for the backend seems like it should be generalized.

briangershon replied:
Sounds good, I'll rename to `price`.

As far as "how long active", should we just run with an "active" boolean, and the Creator can just toggle that off when done? (setting "active" to false would either mean Creator "deleted/inactivated it" or it expired)

briangershon committed:
Rename monthly_price to price. Support deleting
of Service Level accounts.

briangershon committed:
Test SupportLevel validators and ensure only
owners can edit support levels.

Two commented lines for upcoming changes.

briangershon committed:
Receipt now requires SupportLevel, and pulls
amount user pays from SupportLevel and “amount”
is no longer directly passed in.

Update receipt tests to verify.
